File: Task12/Garbage.m
//
//  Garbage.m
//  Task12
//
//  Created by Sergey Plotnikov on 07.07.16.
//  Copyright © 2016 Sergey Plotnikov. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Garbage.h"


@interface Garbage()

@property(nonatomic) NSInteger calories;

@end


@implementation Garbage

@synthesize calories = _calories;

- (instancetype)init
{
    self = [super init];
    
    if (self) {
        _calories = kDefaultGarbageCalories;
    }
    
    return self;
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"Мусор(%ld)", self.calories];
}

@end

File: Task1_2/Garbage.swift
//
//  Garbage.swift
//  Task1_2
//
//  Created by Admin on 23/09/16.
//  Copyright © 2016 Admin. All rights reserved.
//

import Foundation

class Garbage: Creature {
    
    init() {
        super.init(calories: Creature.defaultGarbageCalories)
    }
    
    override var description: String {
        get {
            return "Garbage"
        }
    }
}

